lib/IRGen/GenRecord.h

```diff
@@ -271,6 +271,24 @@ class RecordTypeInfoImpl : public Base,
                                                    fType);
     }
   }
+
+private:
+  // For some types we might need the record's layout metadata
+  bool shouldEmitMetadataRefForLayout(IRGenModule &IGM,
+                                      const CanType canType) const {
+    if (!irgen::isTypeDependent(IGM, canType)) {
+      return false;
+    }
+    if (auto genTuple = dyn_cast<TupleType>(canType)) {
+      for (auto elt : genTuple->getElements()) {
+        if (isTypeDependent(IGM, elt.getType()->getCanonicalType())) {
```

Comment:
Same as above -- if isTypeDependent() returns true for the tuple type, doesn't it mean it has to return true for one of the component types already? Why the additional checks?